src/System.Private.ServiceModel/tests/Scenarios/Binding/Http/BasicHttpBindingTests.cs

```diff
@@ -16,32 +16,32 @@ public static class Binding_Http_BasicHttpBindingTests
     [OuterLoop]
     public static void DefaultSettings_Echo_RoundTrips_String()
     {
-        string variationDetails = "Client:: BasicHttpBinding/DefaultValues\nServer:: BasicHttpBinding/DefaultValues";
+        // *** ASSEMBLE *** \\
+        ChannelFactory<IWcfService> factory = null;
+        IWcfService serviceProxy = null;
         string testString = "Hello";
-        StringBuilder errorBuilder = new StringBuilder();
-        bool success = false;
-
-        BasicHttpBinding binding = new BasicHttpBinding(BasicHttpSecurityMode.None);
+        Binding binding = null;
 
+        // *** ACT *** \\
         try
         {
-            ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
-
-            IWcfService serviceProxy = factory.CreateChannel();
+            binding = new BasicHttpBinding(BasicHttpSecurityMode.None);
+            factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
 
+            serviceProxy = factory.CreateChannel();
             string result = serviceProxy.Echo(testString);
-            success = string.Equals(result, testString);
 
-            if (!success)
-            {
-                errorBuilder.AppendLine(String.Format("    Error: expected response from service: '{0}' Actual was: '{1}'", testString, result));
-            }
+            // *** VALIDATE *** \\
+            Assert.True(result == testString, String.Format("Error: expected response from service: '{0}' Actual was: '{1}'", testString, result));
+
+            // *** GRACEFUL CLEANUP *** \\
+            factory.Close();
+            ((ICommunicationObject)serviceProxy).Close();
         }
-        catch (Exception ex)
+        // *** FORCE CLEANUP *** \\
+        finally
         {
-            errorBuilder.AppendLine(String.Format("    Error: Unexpected exception was caught while doing the basic echo test for variation...\n'{0}'\nException: {1}", variationDetails, ex.ToString()));
+            ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, (ICommunicationObject)factory);
         }
```

Comment:
Were you really forced to cast these explicitly to ICommunicationObject like this/  I would have expected the compiler to just let you pass them in without the cast (because they all implement ICommunicationObject). I'd prefer the shortest possible form.